examples/ecs/event_consumer.rs

```diff
@@ -0,0 +1,65 @@
+use bevy::prelude::*;
+use bevy::{app::Events, core::FixedTimestep};
+
+/// Events are automatically cleaned up after two frames when intialized via `.add_event`.
+/// To bypass this, you can simply add the Events::<T> resource manually.
+/// This is critical when working with systems that read events but do not run every tick,
+/// such as those that operate with a FixedTimeStep run criteria.
+///
+/// When you do so though, you need to be careful to clean up these events eventually,
+/// otherwise the size of your vector of events will grow in an unbounded fashion.
+///
+/// `EventConsumer::<T>` provides a simple interface to do so, clearing all events that it reads
+/// by draining them into a new vector.
+/// You can combine it with other `EventReader`s as long as they read events before,
+/// but only one `EventConsumer` system should be used per event type in most cases
+/// as they will compete for events.
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        // We can't use .add_event or our events will be cleaned up too soon
+        .init_resource::<Events<MyEvent>>()
```

Comment:
I feel like this is a little nuisanced and could be easily missed by a user.
Is there a way to group an `EventConsumer` with `add_event` or warn if they are both added?

Reply:
Hmm. I don't think there's a sensible way to group it. The latter feels useful, but I'm not quite sure how we'd manage that.

Basically we'd need fairly global analysis to get that right; we need to see a) does any system use an EventConsumer and b) was add_event::T ever called.

Reply:
Maybe the answer is to rename `add_event` to be more explicit? I'm not sure what name you'd want there though :/

We can also ramp up the docs on both EventConsumer and add_event some more?

Reply:
Actually: what if we added a `cleanup` field to add_event that explicitly controlled the behavior? I think that would help make things much more explicit and demistify the default behavior.